pkg/gardenadm/cmd/init/options.go

```
@@ -13,15 +15,28 @@ import (
 // Options contains options for this command.
 type Options struct {
 	*cmd.Options
+	// ConfigDir is the path to a directory containing the Gardener configuration files for the init command, i.e.,
+	// files containing resources like CloudProfile, Shoot, etc.
+	ConfigDir string
 }
 
 // ParseArgs parses the arguments to the options.
 func (o *Options) ParseArgs(_ []string) error { return nil }
 
 // Validate validates the options.
-func (o *Options) Validate() error { return nil }
+func (o *Options) Validate() error {
+	if len(o.ConfigDir) == 0 {
+		return fmt.Errorf("must provide a path to a config directory")
+	}
```

Should we check whether the directory actually exists?

---

I don't think this is the right place to check such things - if it doesn't, the `gardenadm init` run will fail early on. Otherwise, we'd also have to add other checks, e.g., if proper resources (e.g., a `Shoot`, `CloudProfile`, etc.) exist in this directory (what would be the logic to keep this outside?).

---

The main reason for checking would be to get a proper error message for cases when the operator simply mistyped the directory. It is more helpful to get something along the lines of "directory does not exist" compared to "must provide a *gardencorev1beta1.CloudProfile resource, but did not find any".
If `fs.WalkDir(...)` already returns a proper error on a non-existing directory that would be enough. I will check this.

---

When using a non-existing directory, the output looks like this:

```
./bin/gardenadm init -d aksmd
Error: failed reading Kubernetes resources from config directory aksmd: failed reading Kubernetes resources from config directory: failed walking directory: stat .: no such file or directory
Usage:
```

I guess that's good enough?

---

Not perfect, but good enough :-) .